### PR TITLE
Spark 3.3 support

### PR DESCRIPTION
**Pull request by SparkSnail**
1. Support running spark.net on Spark3.3.0, Spark3.3.1, Spark3.3.2, Spark3.3.3, Spark3.3.4 environments
2. Add E2E pipeline for Spark3.3.0, Spark3.3.1, Spark3.3.2, Spark3.3.3, Spark3.3.4 environments

### CODE REVIEW

File: src/csharp/Microsoft.Spark.Worker.UnitTest/PayloadWriter.cs [hunk 1]
```diff
@@ -112,6 +112,44 @@ public void Write(Stream stream, TaskContext taskContext)
...
+        {
+            SerDe.Write(stream, taskContext.IsBarrier);
+            SerDe.Write(stream, taskContext.Port);
+            SerDe.Write(stream, taskContext.Secret);
+
+            SerDe.Write(stream, taskContext.StageId);
+            SerDe.Write(stream, taskContext.PartitionId);
+            SerDe.Write(stream, taskContext.AttemptNumber);
+            SerDe.Write(stream, taskContext.AttemptId);
+            SerDe.Write(stream, taskContext.CPUs);
```
Comment (wudanzy) [inline src/csharp/Microsoft.Spark.Worker.UnitTest/PayloadWriter.cs:130]: Please add a comment here to highlight the difference.

**Review by wudanzy — APPROVED** [review]

File: src/csharp/Microsoft.Spark.Worker.UnitTest/PayloadWriter.cs (hunk 1 above)
Comment (SparkSnail) [reply src/csharp/Microsoft.Spark.Worker.UnitTest/PayloadWriter.cs:131]: updated.

**Review by wudanzy — APPROVED** [review]

**Pull request merged**